astropy/modeling/core.py

```diff
@@ -366,7 +366,8 @@ def update_wrapper(wrapper, cls):
                 wrapper.__qualname__ = f'{cls.__qualname__}.{wrapper.__name__}'
 
         if ('__call__' not in members and 'n_inputs' in members and
-                isinstance(members['n_inputs'], int) and members['n_inputs'] > 0):
+                np.issubdtype(type(members['n_inputs']), np.integer) and
+                members['n_inputs'] > 0):
```

Comment:
I would keep this one as a Python int only because this is used often, e.g. by the mapping models, and it's much slower than to check for an int. I can't think of a case where this will be a numpy integer.

Reply:
Done